apps/core/services/externals/fetch_service.py

```
@@ -0,0 +1,367 @@
+                    await session.initialize()
+                    logger.debug(f"MCP 세션 초기화 완료: {url}")
+                    
+                    params = {"url": url}
+                    result = await session.call_tool("fetch_txt", params)
+                    content = getattr(result, "content", None)
+                    
+                    logger.debug(f"MCP 도구 실행 완료: {url}")
+                    return self._process_mcp_result(url, content)
+                    
+        except ConnectionError as e:
+            raise MCPConnectionError(f"MCP 서버 연결 실패: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)